nwplus/nwhacks2019
# Web/admin/applicants page

:tickets: **Ticket(s)**: Closes #235 

---

## :construction_worker: Changes
Mostly implemented, left some smaller items under 'todo' that will be done in the afternoon. In it's current state you can create tags, select specific groups of applicants, and tag/untag them. Pushed it out so that the reviews are less rushed.

**Implemented:**
- `/#/admin/applicants` page
  - ability to view tags
  - ability to select applicants with checkboxes
  - tagging UI
    - can create new tags
    - can select and tag applicants
    - checkboxes include an "indeterminate" state that's shown when only some of the checked applicant's have a specific tag
- `<ApplicantInfo />`: assessment info / full info view for hacker & volunteer
- Added a tie breaker (by timestamp) when sorting by score

**Todo:**
- [x] Show applicantion submission date in `<ApplicantInfo />`
- [x] Only show tag button when applicants are checked
- [x] Move export emails to new toolbar
- [x] Show # of checked applicants
- [x] Filter: select top N (totally forgot about this lol)

Let me know if I've missed anything!

## :flashlight: Testing Instructions
1. `make dev`
2. Go to `/#/admin/applicants`
3. Try creating new tags
3. Try checking + tagging/untagging applicant(s) without any tags
4. Try checking + tagging/untagging applicant(s) with tags
5. Try checking + tagging/untagging applicants with mixed tags. You should see the indeterminate state for the checkboxes if only *some* of the checked applicants have a specific tag.
6. Try cancelling the tag menu (just press out of the modal) and ensure that tags aren't changed for checked applicants
